### ISSUE TITLE
If a singularity eats the arrival shuttle, people will spawn in space

### ISSUE THREAD

**Issue by coiax** [tgstation/tgstation#33758]
Round ID: 80611

People when joining the game will spawn randomly in space and will trigger no arrivalrattle.

**Comment by Cyberboss** [comment]
There should be some sort of handler for a mobile dock having all it's turfs deleted so it can properly react/delete itself

**Comment by Hatfish** [comment]
It seems like the shuttle simply isn't repairing itself at all no matter what, not just when it gets eaten.

**Comment by ninjanomnom** [comment]
...It's not supposed to repair itself?

As for detecting when the shuttle has been fully destroyed, that's a difficult prospect considering shuttles don't necessarily have to be made up of turfs.

A dry run of the movement procedure would be needed to check if anything is going to move.

**Comment by Hatfish** [comment]
It used to repair itself, it still says "Detecting damage to arrivals shuttle, docking for repairs".  The shuttle is supposed to be a safe place to spawn into, so you don't get a situation where 20 or so people are crippled by pressure before the shuttle docks because some space dust punched a hole into the side of the arrivals shuttle.

**Comment by Cyberboss** [comment]
No, it doesn't repair itself. It perma docks at the station and that
message is a prompt for the crew to repair it

On Dec 22, 2017 10:18 PM, "Hatfish" <notifications@github.com> wrote:

It used to repair itself, it still says "Detecting damage to arrivals
shuttle, docking for repairs". The shuttle is supposed to be a safe place
to spawn into, so you don't get a situation where 20 or so people are
crippled by pressure before the shuttle docks because some space dust
punched a hole into the side of the arrivals shuttle.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/tgstation/tgstation/issues/33758#issuecomment-353703942>,
or mute the thread

**Comment by 81Denton** [comment]
Same with the new BoH bomb; players will spawn on seats that fell into the rift in reality.
